src/csvwidget/widget.ts

```diff
@@ -217,6 +217,27 @@ function createDelimiterSwitcherNode(): HTMLElement {
 export
 class CSVWidgetFactory extends ABCWidgetFactory<CSVWidget, DocumentRegistry.IModel> {
   /**
+   * The name of the widget to display in dialogs.
+   */
+  get name(): string {
+    return 'Table';
+  }
+
+  /**
+   * The file extensions the widget can view.
+   */
+  get fileExtensions(): string[] {
+    return ['.csv'];
```

Comment:
I like being able to see in the actual widget class what extensions it will serve. The compromise of passing extensions in via the base class's constructor options seems good.